src/plugins/data/public/query/query_service.ts

```diff
@@ -81,13 +82,21 @@ export class QueryService implements PersistableStateService<QueryState> {
 
   state$!: QueryState$;
 
-  public setup({ storage, uiSettings, nowProvider }: QueryServiceSetupDependencies): QuerySetup {
+  constructor(private minRefreshInterval: number = 5000) {}
+
+  public setup({
+    storage,
+    uiSettings,
+    nowProvider,
+    minRefreshInterval = this.minRefreshInterval,
+  }: QueryServiceSetupDependencies): QuerySetup {
     this.filterManager = new FilterManager(uiSettings);
 
     const timefilterService = new TimefilterService(nowProvider);
     this.timefilter = timefilterService.setup({
       uiSettings,
       storage,
+      minRefreshInterval,
```

Comment:
Nit: Can we move this to the `constructor` of the `TimeFilterService` rather than the `setup` method? I think historically we've tried to keep things inside of `setup` limited to the services/plugins that this service depends on.

Reply:
See https://github.com/elastic/kibana/pull/188881#discussion_r1723794360